src/ldp/utils.py

```diff
@@ -110,3 +113,54 @@ def format_error_details(error: Exception) -> str:
             error_details += f"\nResponse body: {error.response.text}"
 
     return error_details
+
+
+def split_message_transitions(list_of_messages: list[Message]) -> list[list[Message]]:
+    """
+    Break down messages into transitions: [(system, user), (tool request, tool response, env state), ...].
+
+    Blocks end when either:
+    - an EnvStateMessage to ToolRequestMessage transition occurs
+    - a ToolResponseMessage to ToolRequestMessage transition occurs
+
+    Args:
+        list_of_messages: The list of messages to break down.
+
+    Returns:
+        A list of transitions blocks, corresponding to trajectory transition.
+    """
+    if not list_of_messages:
+        return []
```

Comment:
Can we just design this function to be a no-op if `list_of_messages` is an empty list?

Reply:
wait isn't it the same returning `list_of_messages` than `[]` in that case?

Reply:
Feel free to ignore this then